pkg/util/kubernetes/apiserver/metadata_controller_test.go

```diff
@@ -434,9 +421,18 @@ func TestMetadataController(t *testing.T) {
 	_, err = c.Endpoints("default").Create(ep)
 	require.NoError(t, err)
 
-	requireReceive(t, metaController.endpoints, "endpoints")
+	metaController, informerFactory := newFakeMetadataController(client)
+
+	stop := make(chan struct{})
+	defer close(stop)
+	informerFactory.Start(stop)
+	go metaController.Run(stop)
 
-	metadataNames, err = GetPodMetadataNames(node.Name, pod.Namespace, pod.Name)
+	for !metaController.endpointsListerSynced() && !metaController.nodeListerSynced() {
+		time.Sleep(500 * time.Millisecond)
```

Comment:
do we need 500ms here? What's the shortest time it can take to sync?

Reply:
I can drop to 150ms but at 100ms I have some flakiness